annotations/annotations_rw_test.go

```diff
@@ -112,7 +121,9 @@ func TestReadInvalidBodyError(t *testing.T) {
 	rw := NewRW(testClient, s.URL)
 	ctx := tidUtils.TransactionAwareContext(context.Background(), tid)
 	_, _, found, err := rw.Read(ctx, testContentUUID)
-	assert.EqualError(t, err, "invalid character 'i' looking for beginning of object key string")
+	var jsonErr *json.SyntaxError
+	assert.True(t, errors.As(err, &jsonErr))
```

Comment:
`As` will panic if target is nil or not a pointer. Test if the `err` is `nil` first and then use `As`. 
```suggestion
	if (assert.Error(t, err)) {
		assert.True(t, errors.As(err, &jsonErr))
	}
```